### PR TITLE
Layouts

### PR DESCRIPTION
**Pull request by Angham116**
#39  finished layouts

### CODE REVIEW

**Comment by kannanhassouna** [comment]
when you add all of your team as reviewers until that you can't merge your P.R 
@Angham116

File: src/views/layouts/error.hbs [hunk 1]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+   <head>
+      {{>head}}
+      <title> {{title}} </title>
```
Comment (kannanhassouna) [inline src/views/layouts/error.hbs]: I think ``` title ``` and ``` link ``` should be in ``` head ``` partial not here? 
and I think you will pass CSS files as an array?

File: src/views/layouts/error.hbs [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+      <link rel="stylesheet" href="./css/{{css}}.css">
+   </head>
+   <body>
+      {{{body}}
+      <script src="./js/{{js}}.js"></script>
```
Comment (kannanhassouna) [inline src/views/layouts/error.hbs]: I think you will pass js files as an array?

File: src/views/layouts/login_signup.hbs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+   <head>
+      {{>head}}
```
Comment (kannanhassouna) [inline src/views/layouts/login_signup.hbs]: same problem here

File: src/views/layouts/main.hbs [hunk 4]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
```
Comment (kannanhassouna) [inline src/views/layouts/main.hbs:2]: and here

File: src/views/layouts/profile.hbs [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!DOCTYPE html>
+<html lang="en">
+   <head>
+      {{>head}}
```
Comment (kannanhassouna) [inline src/views/layouts/profile.hbs]: and here

**Review by kannanhassouna — CHANGES_REQUESTED** [review]

File: src/views/layouts/error.hbs (hunk 1 above)
Comment (Angham116) [reply src/views/layouts/error.hbs]: but I think I will pass it as an object like : 

```javascript 
res.render('profile',
   {
    js: '.profile.js', css: 'profile.css', layout: 'profile', title: 'Profile | GFC'
  }
);
```
 
 ???  @knanahassouna1

File: src/views/layouts/error.hbs (hunk 2 above)
Comment (Angham116) [reply src/views/layouts/error.hbs]: same as I said above?

File: src/views/layouts/login_signup.hbs (hunk 3 above)
Comment (Angham116) [reply src/views/layouts/login_signup.hbs]: is I will send all ```head``` tag as partial?

File: src/views/layouts/main.hbs (hunk 4 above)
Comment (Angham116) [reply src/views/layouts/main.hbs:2]: how? 
like this:
```javascript
{{{> html}}}
```
 ?!

File: src/views/layouts/error.hbs (hunk 1 above)
Comment (kannanhassouna) [reply src/views/layouts/error.hbs]: yeah you pass it as a object but ``` css & js ``` properties will be arrays 
```
res.render('profile', {
    js: ['.profile.js','.file2.js', '.file3.js'],
    css: [ 'profile.css', '.file2.css', '.file3.css'] , 
    layout: 'profile', 
    title: 'Profile | GFC'
  });
```

File: src/views/layouts/login_signup.hbs (hunk 3 above)
Comment (kannanhassouna) [reply src/views/layouts/login_signup.hbs]: yeah

File: src/views/layouts/main.hbs (hunk 4 above)
Comment (kannanhassouna) [reply src/views/layouts/main.hbs:2]: you should put all these stuff in head partial

**Review by kannanhassouna — APPROVED** [review]

**Review by Fatmasiam — APPROVED** [review]

**Review by ahmedalmi95 — APPROVED** [review]

File: src/views/layouts/main.hbs (hunk 4 above)
Comment (Anies12) [reply src/views/layouts/main.hbs:2]: remove one curly bracket before the body

**Comment by kannanhassouna** [comment]
:+1:  Good Job

**Review by Anies12 — CHANGES_REQUESTED** [review]
modify the layouts 
the number of the beginning brackets is not the same as the ending ones

File: src/views/layouts/main.hbs (hunk 4 above)
Comment (Angham116) [reply src/views/layouts/main.hbs:2]: done.

**Review by Anies12 — APPROVED** [review]

**Pull request merged**